vdj_ann/src/vdj_features.rs

```diff
@@ -569,9 +573,11 @@ pub fn fr3_start(aa: &[u8], chain_type: &str, verbose: bool) -> Option<usize> {
         ];
 
         // Score positions.
-
+        if cdr3_start < 35 {
+            return None;
+        }
         let mut score_pos = Vec::<(usize, usize)>::with_capacity(38 - 35 + 1);
-        for j in cdr3_start - 38..=cdr3_start - 35 {
+        for j in cdr3_start.saturating_sub(38)..=cdr3_start - 35 {
```

Comment:
It's probably better to return `None` if `cdr3_start < 38`. Such a situation is very unlikely for a full length VDJ transcript, but ensures that the `FR3` start computed based on a given contig is the same irrespective of the amount of prefix clipped in the contig

Reply:
^Similarly for other cases

Reply:
Ok cool, that makes sense to me.  I will update the check conditionals and remove the saturating subtractions.  Can you confirm that this is the right thing to do for all cases?  I'm mainly wondering about the one case that is already using saturating subtraction for both the start and end, though I suspect this is simply a bug.

Reply:
Yep, let's do this for all cases. The saturating sub looks like a bug indeed!

Reply:
Done